lib/season_stat_coach.rb

```diff
@@ -0,0 +1,57 @@
+require_relative 'game_collection'
+require_relative 'game_team_collection'
+require_relative 'team_collection'
+
+class SeasonStatCoach
+
+  def initialize(game_file_path, team_file_path, game_team_file_path)
+    @game_collection = GameCollection.new(game_file_path)
+    @team_collection = TeamCollection.new(team_file_path)
+    @game_team_collection = GameTeamCollection.new(game_team_file_path)
+    @season_list = @game_collection.get_all_seasons
+  end
+
+  def get_season_game_teams(season)
+    @game_team_collection.game_team_list.find_all do |game_team|
+      game_team.game_id[0..3] == season[0..3]
+    end
+  end
+
+  def coaches_by_season(season)
+    get_season_game_teams(season).map do |game|
+      game.head_coach
+    end.uniq
+  end
+
+  def get_coach_wins_by_season(coach, season)
+    get_season_game_teams(season).find_all do |game_team|
+      game_team.head_coach == coach && game_team.result == "WIN"
+    end.length
+  end
+
+  def get_total_coach_games_by_season(coach, season)
+    get_season_game_teams(season).find_all do |game_team|
+      game_team.head_coach == coach
+    end.length
+  end
+
+  def coach_win_percentage_by_season(coach, season)
+    win_total = get_coach_wins_by_season(coach, season).to_f
+    total_games = get_total_coach_games_by_season(coach, season)
+    ((win_total / total_games) * 100).round(2)
```

Comment:
i might be able to remove this calculation because i only need the value to compare not the actual percent

Reply:
the * 100 part i mean